### ISSUE TITLE
one region is always nan

### ISSUE THREAD

**Issue by Francis-Xia** [AlexeyAB/darknet#2321]
![info_train](https://user-images.githubusercontent.com/8150586/51970743-a014b100-24b2-11e9-8af8-75ab24413ebf.png)

**Comment by Francis-Xia** [comment]
my cfg:

`[net]
# Testing
# batch=1
# subdivisions=1
# Training
batch=128
subdivisions=2
width=320
height=160
channels=3
momentum=0.9
decay=0.0005
angle=10
saturation = 1.5
exposure = 1.5
hue=0.1

learning_rate=0.001
burn_in=-1
max_batches=13200
policy=steps
steps=600, 1200, 1800, 2400, 3000, 3600, 4200, 4800, 5400, 6000, 6600, 7200, 7800, 8400, 9000, 9600, 10200, 10800, 11400, 12000
scales=0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75, 0.75

# 0
# ConvNd_1
[convolutional]
batch_normalize=1
filters=16
size=3
stride=2
pad=1
activation=leaky

# 1
# ConvNd_2
[convolutional]
batch_normalize=1
filters=32
size=2
stride=2
pad=0
activation=leaky

# 2
# ConvNd_3
[convolutional]
batch_normalize=1
filters=32
size=2
stride=1
pad=0
activation=leaky

# 3
# ConvNd_4
[convolutional]
batch_normalize=1
filters=32
size=2
stride=1
pad=1
activation=leaky

# 4
# ConvNd_5
[route]
layers = 0

# 5
[convolutional]
batch_normalize=1
filters=32
size=2
stride=2
pad=0
activation=leaky

# 6
# Add1_1
[shortcut]
from=3
activation=leaky

# 7
# ConvNd_6
[convolutional]
batch_normalize=1
filters=32
size=2
stride=2
pad=0
activation=leaky

# 8
# ConvNd_7
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=0
activation=leaky

# 9
# ConvNd_8
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 10
# ConvNd_9
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 11
# Cat_1
[route]
layers = 9, 10

# 12
# ConvNd_10
[route]
layers = 6

# 13
[convolutional]
batch_normalize=1
filters=64
size=2
stride=2
pad=0
activation=leaky

# 14
# Add1_2
[shortcut]
from=11
activation=leaky

# 15
# ConvNd_11
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=0
activation=leaky

# 16
# ConvNd_12
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 17
# ConvNd_13
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 18
# Cat_2
[route]
layers = 16,17

# 19
# ConvNd_14
[route]
layers=14

# 20
[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=0
activation=leaky

# 21
# Add1_3
[shortcut]
from = 18
activation=leaky

# 22
# ConvNd_15
[convolutional]
batch_normalize=1
filters=64
size=2
stride=2
pad=0
activation=leaky

# 23
# ConvNd_16
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=0
activation=leaky

# 24
# ConvNd_17
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 25
# ConvNd_18
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 26
# Cat_3
[route]
layers = 24, 25

# 27
# ConvNd_19
[route]
layers = 21

# 28
[convolutional]
batch_normalize=1
filters=64
size=2
stride=2
pad=0
activation=leaky

# 29
# Add1_4
[shortcut]
from=26
activation=leaky

# 30
# ConvNd_20
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=0
activation=leaky

# 31
# ConvNd_21
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 32
# ConvNd_22
[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

# 33
# Cat_4
[route]
layers=31,32

# 34
# ConvNd_23
[route]
layers=29

#35
[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=0
activation=leaky

# 36
# Add1_5
[shortcut]
from=33
activation=leaky

# 37
# ConvNd_24
[convolutional]
batch_normalize=1
filters=64
size=2
stride=2
pad=0
activation=leaky

# 38
# ConvNd_25
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=0
activation=leaky

# 39
# ConvNd_26
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 40
# ConvNd_27
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 41
# Cat_5
[route]
layers=39,40

# 42
# ConvNd_28
[route]
layers=36

# 43
[convolutional]
batch_normalize=1
filters=128
size=2
stride=2
pad=0
activation=leaky

# 44
# Add1_6
[shortcut]
from=41
activation=leaky

# 45
# ConvNd_29
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=0
activation=leaky

# 46
# ConvNd_30
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 47
# ConvNd_31
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 48
# Cat_6
[route]
layers=46,47

# 49
# ConvNd_32
[route]
layers=44

# 50
[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=0
activation=leaky

# 51
# Add1_7
[shortcut]
from=48
activation=leaky

# 52
# ConvNd_33
[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=0
activation=leaky

# 53
[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

# 54
[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=0
activation=leaky

# 55
[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

# 56
[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=0
activation=leaky

# 57
[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

# 58
[convolutional]
size=1
stride=1
pad=1
filters=24
activation=linear

# 59
[yolo]
mask = 9,10,11
anchors = 7,3, 6,5, 7,6,   14,3, 14,5, 9,8,   24,5, 26,11, 44,8,   68,15, 122,29, 220,82
classes=3
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=1
max=200
focal_loss=1

# 60
[route]
layers = 56

# 61
# PixelShuffle
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 62
[upsample]
stride=2

# 63
[route]
layers = 62,36

# 64
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=1
activation=leaky

# 65
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 66
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=1
activation=leaky

# 67
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 68
[convolutional]
batch_normalize=1
filters=32
size=1
stride=1
pad=1
activation=leaky

# 69
[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

# 70
[convolutional]
size=1
stride=1
pad=1
filters=24
activation=linear

# 71
[yolo]
mask = 6,7,8
anchors = 7,3, 6,5, 7,6,   14,3, 14,5, 9,8,   24,5, 26,11, 44,8,   68,15, 122,29, 220,82
classes=3
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=1
max=200
focal_loss=1

# 72
[route]
layers = 68

# 73
[convolutional]
batch_normalize=1
filters=8
size=1
stride=1
pad=1
activation=leaky

# 74
[upsample]
stride=2

# 75
[route]
layers = 74, 21

# 76
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 77
[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=32
activation=leaky

# 78
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 79
[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=32
activation=leaky

# 80
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 81
[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=32
activation=leaky

# 82
[convolutional]
size=1
stride=1
pad=1
filters=24
activation=linear

# 83
[yolo]
mask = 3,4,5
anchors = 7,3, 6,5, 7,6,   14,3, 14,5, 9,8,   24,5, 26,11, 44,8,   68,15, 122,29, 220,82
classes=3
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=1
max=200
focal_loss=1

# 84
[route]
layers = 80

# 85
[convolutional]
batch_normalize=1
filters=4
size=1
stride=1
pad=1
activation=leaky

# 86
[upsample]
stride=2

# 87
[route]
layers = 86, 6

# 88
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 89
[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=32
activation=leaky

# 90
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 91
[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=32
activation=leaky

# 92
[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

# 93
[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=32
activation=leaky

# 94
[convolutional]
size=1
stride=1
pad=1
filters=24
activation=linear

# 95
[yolo]
mask = 0,1,2
anchors = 7,3, 6,5, 7,6,   14,3, 14,5, 9,8,   24,5, 26,11, 44,8,   68,15, 122,29, 220,82
classes=3
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=1
max=200
focal_loss=1
`

**Comment by Francis-Xia** [comment]
In yolo layer, num should be 12, but 9.
So, I will close the issue.
